mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala

```diff
@@ -116,6 +135,9 @@ object AssociationRules {
     @Since("1.5.0")
     def confidence: Double = freqUnion.toDouble / freqAntecedent
```

Comment:
I suppose this `.toDouble` is redundant actually!

Reply:
yes, it is not related to this PR but I can  remove it.